### PR TITLE
swiching reconciliation from composition to composite

### PR DESCRIPTION
**Pull request by wejdross**
## Summary

* Short summary of what's included in the PR
* Give special note to breaking changes

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ type probeManager interface {
 	StopProbe(p probes.ProbeInfo)
 }
 
+//+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=xvshnpostgresqls,verbs=get;list;watch
+//+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=xvshnpostgresqls/status,verbs=get
 //+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=vshnpostgresqls,verbs=get;list;watch
```
Comment (Kidswiss) [inline pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go:58]: Do we actually still need those permissions after your change?

File: pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go [hunk 2]
```diff
@@ -102,7 +104,7 @@ func (r *VSHNPostgreSQLReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		}
 
 		// Create a pobe that will always fail
-		probe, err = probes.NewFailingPostgreSQL(vshnpostgresqlsServiceKey, inst.Name, inst.Namespace)
+		probe, err = probes.NewFailingPostgreSQL(vshnpostgresqlsServiceKey, inst.Name, inst.ObjectMeta.Labels["crossplane.io/claim-namespace"])
```
Comment (Kidswiss) [inline pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go:107]: You could put the label key in a constant. Maybe it's actually already in a common/util package somewhere...

File: pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go [hunk 3]
```diff
@@ -113,20 +115,29 @@ func (r *VSHNPostgreSQLReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return res, nil
 }
 
-func (r VSHNPostgreSQLReconciler) fetchProberFor(ctx context.Context, inst *vshnv1.VSHNPostgreSQL) (probes.Prober, error) {
+func (r VSHNPostgreSQLReconciler) fetchProberFor(ctx context.Context, inst *vshnv1.XVSHNPostgreSQL) (probes.Prober, error) {
+	instance := &vshnv1.XVSHNPostgreSQL{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      inst.ObjectMeta.Labels["crossplane.io/claim-name"],
```
Comment (Kidswiss) [inline pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go:121]: Same here, use a constant.

File: pkg/sliexporter/probes/postgresql_test.go [hunk 4]
```diff
@@ -29,14 +29,18 @@ func TestPostgreSQL_Probe(t *testing.T) {
 	require.NoError(t, err)
 	defer pool.Purge(res)
 
+	// race condition exists on my machine, therefore sleep to wait for postgres to be ready
+	// there is unfortunately no way to check if postgres is ready using dockertest
+	time.Sleep(3 * time.Second)
```
Comment (Kidswiss) [inline pkg/sliexporter/probes/postgresql_test.go:34]: AFAIK you can get a docker client and list the containers via `pool.Client.ListContainers(opts docker.ListContainersOptions)`. Not sure what it exactly it returns though, but I guess there should be something to determine the readiness of the container or postgresql itself.

Also there's a `pool.Retry()` function which will retry the function you pass to it.

Combining those should give a more robust way to wait for it to be ready.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Looks good, just some minor issues.

But does it solve the actual issue we have? 

Please also add some more information in the description, why we change it.
I just recently found something that was confusing and was not able to figure out WHY it was changed...
So future us will thank you :)

**Comment by wejdross** [comment]
@Kidswiss @TheBigLee  I'm closing this branch and moving to newer one: https://github.com/vshn/appcat/pull/79
because our Project Syn doesn't like special characters in branch names. After I apply suggestions from this branch, I'll invite You both to review

**Pull request closed without merging**